runtime/src/snapshot_utils.rs

```diff
@@ -833,6 +833,32 @@ where
     Ok(())
 }
 
+/// To allow generating a bank snapshot directory with full state information, we need to
+/// hardlink account appendvec files from the runtime operation directory to a snapshot
+/// hardlink directory.  This is to create the run/ and snapshot sub directories for an
+/// account_path provided by the user.  These two sub directories are on the same file
+/// system partition to allow hard-linking.
+pub fn create_accounts_run_and_snapshot_dirs(
+    account_dir: impl AsRef<Path>,
+) -> std::io::Result<(PathBuf, PathBuf)> {
+    let run_path = account_dir.as_ref().join("run");
+    let snapshot_path = account_dir.as_ref().join("snapshot");
+    if run_path.try_exists().is_err() || snapshot_path.try_exists().is_err() {
```

Comment:
When it is a symlink, Ok(true) or Ok(false) tells if the symlink destination is broken or not.  We know here it is not symlink, but a directory.  Is it still needed?  If we really want to be strict, maybe we could use is_dir()?

Reply:
Would we get an error later on if (for some reason) there was a `run` file instead of a directory?

Reply:
changed to is_dir.  It is more strict now.  not only it has to exist, but also should be a dir.  otherwise the whole account_path will be removed and recreated.

Reply:
ignore my previous comment - I was reading it the backwards. `is_dir` is the better choice.